### PR TITLE
Keep original file names when reading files from s3 Issue #611

### PR DESCRIPTION
**Pull request by utility-aagrawal**
Updated the s3 loader to keep the original filenames. Currently, the loader assigns files temporary names which is not useful for RAG use cases where you want to present your users with relevant sources. Related issue #611

### CODE REVIEW

**Comment by utility-aagrawal** [comment]
@logan-markewich , can you please review and merge this request?

File: llama_hub/s3/base.py [hunk 1]
```diff
@@ -92,15 +92,17 @@ def load_data(self) -> List[Document]:
 
         with tempfile.TemporaryDirectory() as temp_dir:
             if self.key:
+                filename = Path(self.key).name
                 suffix = Path(self.key).suffix
-                filepath = f"{temp_dir}/{next(tempfile._get_candidate_names())}{suffix}"
+                filepath = f"{temp_dir}/{filename}"
+                # filepath = f"{temp_dir}/{next(tempfile._get_candidate_names())}{suffix}"
```
Comment (nerdai) [inline llama_hub/s3/base.py:98]: Can you remove these commented lines if not necessary

File: llama_hub/s3/base.py [hunk 2]
```diff
@@ -113,7 +115,8 @@ def load_data(self) -> List[Document]:
                         continue
 
                     filepath = (
-                        f"{temp_dir}/{next(tempfile._get_candidate_names())}{suffix}"
+                        f"{temp_dir}/{filename}"
+                        # f"{temp_dir}/{next(tempfile._get_candidate_names())}{suffix}"
```
Comment (nerdai) [inline llama_hub/s3/base.py:119]: Please remove this as well :)

**Review by nerdai — COMMENTED** [review]
Looks good, just minor comments. BTW, the only file that you modified is `s3/base.py`, right? I'm assuming the other files were modified from `make lint` and `make format`?

**Comment by utility-aagrawal** [comment]
@nerdai Yes, the only file is modified is s3/base.py. Removed the commented lines. Let me know in case of any further questions. Thanks!

**Comment by nerdai** [comment]
> @nerdai Yes, the only file is modified is s3/base.py. Removed the commented lines. Let me know in case of any further questions. Thanks!

Thanks! merging now :)

**Pull request merged**